src/dlgTriggerEditor.cpp

```diff
@@ -6766,6 +6766,55 @@ void dlgTriggerEditor::changeView(EditorViewType view)
     checkBox_displayAllVariables->setVisible(view == EditorViewType::cmVarsView);
 
     mpExportAction->setEnabled(view != EditorViewType::cmVarsView);
+
+    switch (mCurrentView) {
+    case EditorViewType::cmTriggerView:
+        mAddItem->setIconText(tr("Add Trigger"));
+        mAddGroup->setIconText(tr("Add Trigger Group"));
+        mDeleteItem->setIconText(tr("Delete Trigger"));
+        mSaveItem->setIconText(tr("Save Trigger"));
+        break;
+    case EditorViewType::cmTimerView:
+        mAddItem->setIconText(tr("Add Timer"));
+        mAddGroup->setIconText(tr("Add Timer Group"));
+        mDeleteItem->setIconText(tr("Delete Timer"));
+        mSaveItem->setIconText(tr("Save Timer"));
+        break;
+    case EditorViewType::cmAliasView:
+        mAddItem->setIconText(tr("Add Alias"));
+        mAddGroup->setIconText(tr("Add Alias Group"));
+        mDeleteItem->setIconText(tr("Delete Alias"));
+        mSaveItem->setIconText(tr("Save Alias"));
+        break;
+    case EditorViewType::cmScriptView:
+        mAddItem->setIconText(tr("Add Script"));
+        mAddGroup->setIconText(tr("Add Script Group"));
+        mDeleteItem->setIconText(tr("Delete Script"));
+        mSaveItem->setIconText(tr("Save Script"));
+        break;
+    case EditorViewType::cmActionView:
+        mAddItem->setIconText(tr("Add Button"));
+        mAddGroup->setIconText(tr("Add Button Group"));
+        mDeleteItem->setIconText(tr("Delete Button"));
+        mSaveItem->setIconText(tr("Save Button"));
+        break;
+    case EditorViewType::cmKeysView:
+        mAddItem->setIconText(tr("Add Key"));
+        mAddGroup->setIconText(tr("Add Key Group"));
+        mDeleteItem->setIconText(tr("Delete Key"));
+        mSaveItem->setIconText(tr("Save Key"));
+        break;
+    case EditorViewType::cmVarsView:
+        mAddItem->setIconText(tr("Add Variable"));
+        mAddGroup->setIconText(tr("Add Variable Group"));
+        mDeleteItem->setIconText(tr("Delete Variable"));
+        mSaveItem->setIconText(tr("Save Variable"));
+        break;
+    default:
+        qDebug() << "ERROR: dlgTriggerEditor::changeView() undefined view";
```

Comment:
:warning: Unfortunately I do see this being hit on an irregular basis - but never bothered to investigate - given the effect it will now have (not changing the texts perhaps when they should be) perhaps we should do so before committing to this...

Just recently (last day or so) I was actually on-line and in WoTMUD to capture some game replay data for some changes I want to make elsewhere in the codebase and I noticed that some editor changes were not "sticking" - i.e. I'd change something, save it, switch away, and then switch back, and found my change had gone away - and I think this was associated with this error message. Someone will need to investigate...

Reply:
A good idea to log it as an issue.